app/main.py

```diff
@@ -1,3 +1,80 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    @staticmethod
+    def check_value(value: Distance | int | float) -> int | float:
+        return (
+            value if isinstance(value, (int, float))
+            else value.km
+        )
```

Comment:
The `check_value` method should explicitly handle both `Distance` and numeric types. Currently, it assumes that if the value is not an `int` or `float`, it must be a `Distance`. Consider adding a check to ensure the value is indeed a `Distance` before accessing its `km` attribute.